docs/reference/security/sts-login.md

```diff
@@ -0,0 +1,145 @@
+title: Secure Token Service (STS) Authentication for lakeFS
+description: Authenticate with lakeFS using Secure Token Service (STS) by leveraging a remote authenticator. This feature enables integration with Identity Providers (IdPs) for secure and efficient user authentication.
+grand_parent: Reference
+parent: Security
+redirect_from:
+  - /reference/remote-authenticator.html
+---
+
+# STS Login
+
+{: .d-inline-block }
+<a style="color: white;" href="#sso-for-lakefs-cloud">lakeFS Cloud</a>
+{: .label .label-green }
+
+{: .d-inline-block }
+<a style="color: white;" href="#sso-for-lakefs-enterprise">lakeFS Enterprise</a>
+{: .label .label-purple }
+
+{: .note}
+> STS Login is available in lakeFS Cloud and lakeFS Enterprise. currently only tested on lakeFS Enterprise.
+
+Secure Token Service (STS) authentication in lakeFS allows users to authenticate via an Identity Provider (IdP), using OIDC Authentication workflow.
+This document outlines the process, from setting up the STS authentication flow to integrating with lakeFS through the [high-level Python SDK](../../integrations/python.md).
+
+## Architecture
+
+{: .note}
+> The architecture does include the steps to generate the code using the remote authenticator. this should be done prior to the STS login flow.
+> There is a sample implementation to generate the code, redirect_uri and state at the end of this document.
+
+The STS authentication flow involves several components that facilitate secure communication between the lakeFS client, lakeFS server, the remote authenticator, and the IdP.
+
+```mermaid
+sequenceDiagram
+    participant A as lakeFS Client
+    participant B as lakeFS Server
+    participant C as Remote Authenticator
+    participant D as IdP
+    A->>B: Call STS login endpoint
+    B->>C: POST idp code state and redirect uri
+    C->>D: IdP request
+    D->>C: IdP response
+    C->>B: Auth response
+    B->>A: auth JWT
+```
+- lakeFS Client: Initiates the authentication process by providing IdP credentials.
+- lakeFS Server: Facilitates the authentication request between the client and the remote authenticator.
+- Remote Authenticator: Acts as a bridge between lakeFS and the IdP, handling credential validation.
+- IdP (Identity Provider): Validates the provided credentials and returns the authentication status.
+
+## Configuration
+
+To enable STS authentication, configure your lakeFS instance with the appropriate endpoint to the external Authentication Service.
+
+
+```yaml
+
+auth:
+    authentication_api:
+        endpoint: <url-to-remote-authenticator-endpoint>
+
+
+```
+
+- `auth.authentication_api.endpoint` `(string: https://external.authentication-service/api/v1)` - URL to external Authentication Service described at [authentication.yml](https://github.com/treeverse/lakeFS/blob/master/api/authentication.yml)
+
+Ensure you replace <url-to-remote-authenticator-endpoint> with the actual URL of your Authentication Service.
+
+
+## STS Login with high level python client
```

Comment:
IMO the order of actions a user needs to take is not 100% clear. 
Can you create a numbered list of steps for configuration to a successful login?

Reply:
Sounds like a good idea, I'm not really sure how to do that, Add a previous section with steps and links?
One concern I do have with the steps is that (I'm assuming) the user configuring and the user using the log in with the high level python client are not the same user, also the configuration is a one time thing and the login is not, @talSofer  WDYT? is it even a concern?

Reply:
> One concern I do have with the steps is that (I'm assuming) the user configuring and the user using the log in with the high level python client are not the same user

that's a good point. 

If we keep Configuration separate from Usage, how many steps there are before you can run the Python snippet?

Reply:
Only the steps for generating the `code` and `state` which is up to the client to configure in the way they want. so basically there are three separate independent (almost) steps
- configuration
- generating the `code` and `state` - which is up to the client but we have provided a sample code of how to do that
- generate client with the python wrapper (two lines of code)